### PR TITLE
Add documentation for scripts and service calls with response data

### PR DESCRIPTION
**Pull request by allenporter**
## Proposed change

Add documentation for how to use service calls with response data in a script.

This uses an example of calling the calendar API, and is somewhat duplicated in https://github.com/home-assistant/home-assistant.io/pull/27934 which I think makes sense given this is a new use case without many other examples. (Perhaps we can pick a simpler service in the future)

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/94757 https://github.com/home-assistant/core/pull/94759
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/scripts/service-calls.markdown [hunk 1]
```diff
@@ -125,6 +125,57 @@ data: >
 
 {% endraw %}
 
+### Use templates to handle response data
+
+Some services may respond with data that can be used in automations. Services response data
```
Comment (c0ffeeca7) [inline source/_docs/scripts/service-calls.markdown]:
```suggestion
Some services may respond with data that can be used in automation. This data is called _service response data_. Services response data
```

File: source/_docs/scripts/service-calls.markdown [hunk 2]
```diff
@@ -125,6 +125,57 @@ data: >
...
+is typically used for data that is dynamic or large which may not be appropriate for entity state.
```
Comment (c0ffeeca7) [inline source/_docs/scripts/service-calls.markdown]:
```suggestion
is typically used for data that is dynamic or large and which may not be suited for use for entity state.
```

File: source/_docs/scripts/service-calls.markdown [hunk 3]
```diff
@@ -125,6 +125,57 @@ data: >
...
+Examples may include things like upcoming calendar events for the next week or detailed driving directions.
```
Comment (c0ffeeca7) [inline source/_docs/scripts/service-calls.markdown]:
```suggestion
Examples of service response data are upcoming calendar events for the next week or detailed driving directions.
```

File: source/_docs/scripts/service-calls.markdown [hunk 4]
```diff
@@ -125,6 +125,57 @@ data: >
...
+
+Templates can also be used for handling response data. The service call can specify
+`response_variable` which is the name a [variable](https://www.home-assistant.io/docs/scripts/#variables)
```
Comment (c0ffeeca7) [inline source/_docs/scripts/service-calls.markdown]:
```suggestion
a `response_variable`. This is the [variable](https://www.home-assistant.io/docs/scripts/#variables)
```

File: source/_docs/scripts/service-calls.markdown [hunk 5]
```diff
@@ -125,6 +125,57 @@ data: >
...
+that will contain the response data. This example calls a service and sets the response is stored in
```
Comment (c0ffeeca7) [inline source/_docs/scripts/service-calls.markdown]:
```suggestion
that contains the response data. You can define any name for your `response_variable`. This example calls a service and stores the response in 
```

File: source/_docs/scripts/service-calls.markdown [hunk 6]
```diff
@@ -125,6 +125,57 @@ data: >
...
+the variable `agenda`
```
Comment (c0ffeeca7) [inline source/_docs/scripts/service-calls.markdown]:
```suggestion
the variable called `agenda`.
```

File: source/_docs/scripts/service-calls.markdown [hunk 7]
```diff
@@ -125,6 +125,57 @@ data: >
...
+```
+
+{% endraw %}
+
+You may then use the response data in the variable `agenda` in another action
+in the same script. The example below sends a notification using the response
+data.
+
+<div class='note'>
+The service call data fields depend on which notification service is used.
```
Comment (c0ffeeca7) [inline source/_docs/scripts/service-calls.markdown]:
```suggestion
Which data fields can be used in a service call depends on the type of notification service that is used.
```

**Review by c0ffeeca7 — CHANGES_REQUESTED** [review]
Some rephrasing suggestions. Let me know what you think.

File: source/_docs/scripts/service-calls.markdown [hunk 8]
```diff
@@ -125,6 +125,57 @@ data: >
...
+</div>
+
+{% raw %}
+
+```yaml
+service: notify.gmail_com
+data:
+  target: gduser1@workspacesamples.dev
+  title: Daily agenda for {{ now().date() }}
```
Comment (frenck) [inline source/_docs/scripts/service-calls.markdown]:
```suggestion
  target: "gduser1@workspacesamples.dev"
  title: "Daily agenda for {{ now().date() }}"
```

File: source/_docs/scripts/service-calls.markdown [hunk 9]
```diff
@@ -125,6 +125,57 @@ data: >
...
 {% endraw %}
 
+### Use templates to handle response data
+
+Some services may respond with data that can be used in automation. This data is called _service response data_. Services response data
+is typically used for data that is dynamic or large and which may not be suited for use for entity state.
+Examples of service response data are upcoming calendar events for the next week or detailed driving directions.
+
+Templates can also be used for handling response data. The service call can specify
+a `response_variable`. This is the [variable](https://www.home-assistant.io/docs/scripts/#variables)
```
Comment (frenck) [inline source/_docs/scripts/service-calls.markdown]:
```suggestion
a `response_variable`. This is the [variable](/docs/scripts/#variables)
```

**Review by frenck — DISMISSED** [review]
LGTM, thanks @allenporter 👍 

../Frenck

File: source/_docs/scripts/service-calls.markdown [hunk 10]
```diff
@@ -125,6 +125,57 @@ data: >
 
 {% endraw %}
 
+### Use templates to handle response data
+
+Some services may respond with data that can be used in automation. This data is called _service response data_. Services response data
```
Comment (c0ffeeca7) [inline source/_docs/scripts/service-calls.markdown]:
```suggestion
Some services may respond with data that can be used in automation. This data is called _service response data_. Service response data
```

File: source/_docs/scripts/service-calls.markdown [hunk 11]
```diff
@@ -125,6 +125,57 @@ data: >
...
+is typically used for data that is dynamic or large and which may not be suited for use for entity state.
```
Comment (c0ffeeca7) [inline source/_docs/scripts/service-calls.markdown]:
```suggestion
is typically used for data that is dynamic or large and which may not be suited for use in entity state.
```

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @allenporter :+1:

**Review by frenck — APPROVED** [review]

**Pull request merged**